### PR TITLE
Add OccupancyGroup monitoring and export battery status from individu…

### PR DESCRIPTION
**Pull request by achatham**
…al sensors

### CODE REVIEW

**Comment by JonGilmore** [comment]
@achatham do you pylutron for homeassistant? If this PR gets accepted, do you have plans to extend functionality in HASS?

**Comment by achatham** [comment]
That's my intention, yes.

**Comment by JonGilmore** [comment]
> That's my intention, yes.

sweet! Do you have a HASS PR ready based on these changes? I'd love to test it out if so!

**Comment by achatham** [comment]
I had something a while ago but reverted it. It was pretty straightforward
to do. Been waiting for a while to get this approved :)

The other PR I had adds the keypad button location, which I was hoping to
use to make a lutron keypad button act as a binary input in HA, with the
keypad owning the state.

**Comment by dummptyhummpty** [comment]
@achatham Does that work for Pico remotes or only the hard wired keypads?

**Comment by achatham** [comment]
That code would be the keypads. I think you could use the remotes by
binding them to a virtual action in the lutron config and then getting that
from the telnet protocol. But I'm not sure if pylutron supports that yet
and I haven't used the remotes

**Comment by benjimatt** [comment]
This is literally the last part of the RA2 protocol I need integrated.  I just want to add conditional logic without purchasing HWQS.

**Comment by thecynic** [comment]
I'll look at this PR again this afternoon.

File: pylutron/__init__.py [hunk 1]
```diff
@@ -843,16 +843,202 @@ def handle_update(self, args):
...
+  # Values from ?HELP,?DEVICE,22
+  UNKNOWN = 0
+  BATTERY = 1
+  EXTERNAL = 2
+
+  
+class BatteryStatus(Enum):
+  """Enum values representing battery state, reported by queries to
+  battery-powered devices."""
+  UNSET = -1
```
[the remark this answers is not in the document]
Comment (thecynic) [reply pylutron/__init__.py]: I can rework that if you like before merging, in case you have more git problems :)

**Comment by thecynic** [comment]
@achatham please take a look, I pushed 2 patches to your PR. One resolving my comments (I hope that works for you) and the other fixing a bug in setting _last_update. PTAL.

**Comment by achatham** [comment]
LGTM

**Comment by JonGilmore** [comment]
@achatham any chance you've been able to put back together that PR for home assistant? I'm happy to help beta test it prior to pushing to the main line.

**Comment by achatham** [comment]
I did months ago but will reverted it. It didn't take very long so I'll try
again this weekend. Will probably leave out the batter stuff for now

**Comment by JonGilmore** [comment]
Sweet, thanks! Let me know if I can help out

**Pull request merged**